src/datasets/arrow_reader.py

```diff
@@ -319,14 +341,14 @@ def __init__(self, path: str, info: Optional["DatasetInfo"]):
         super().__init__(path, info)
         self._filetype_suffix = "parquet"
 
-    def _get_dataset_from_filename(self, filename_skip_take):
+    def _get_dataset_from_filename(self, filename_skip_take, in_memory=False):
         """Returns a Dataset instance from given (filename, skip, take)."""
         filename, skip, take = (
             filename_skip_take["filename"],
             filename_skip_take["skip"] if "skip" in filename_skip_take else None,
             filename_skip_take["take"] if "take" in filename_skip_take else None,
         )
-        pa_table = pa.parquet.read_table(filename, memory_map=True)
+        pa_table = pa.parquet.read_table(filename, memory_map=not in_memory)
```

Comment:
```suggestion
        pa_table = pa.parquet.read_table(filename, memory_map=True)
```
The `memory_map` of the parquet reader of pyarrow is only to improve the performance of the parquet file loading, but in the end it will still be loaded in memory. See more details in the doc of [pyarrow.parquet.read_table](https://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html)

Reply:
Thanks @lhoestq for pointing this out. Now you make me doubt... Let me check this! ;)

Reply:
Good point, @lhoestq! I have tested it and you are right: the total_allocated_bytes always increase whether memory_map is True or False.